app/models/spree/calculator_decorator.rb

```diff
@@ -9,7 +9,7 @@ def line_items_for(object)
         [object]
       elsif object.respond_to? :line_items
         object.line_items
-      elsif object.order.present?
+      elsif object.respond_to? :order
```

Comment:
it's not equivalent, there will be times where order will return nil and in that case we better return the object and not the nil order.
So maybe better to have the following?
```suggestion
      elsif object.respond_to? :order && object.order.present?
```

I am not sure you have seen but this is partially tested in weight_spec. I think it's very good we are adding tests for this specific class.

Reply:
Yeah, it seems like this method is called from various different places, both in OFN code for calculators, and elsewhere in Spree code.

Reply:
Done!